File: rb/features.rb
# TemporaryEmailApi2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TemporaryEmailApi2Features
  def self.make_feature(name)
    case name
    when "base"
      TemporaryEmailApi2BaseFeature.new
    when "test"
      TemporaryEmailApi2TestFeature.new
    else
      TemporaryEmailApi2BaseFeature.new
    end
  end
end

File: py/features.py
# TemporaryEmailApi2 SDK feature factory

from feature.base_feature import TemporaryEmailApi2BaseFeature
from feature.test_feature import TemporaryEmailApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: TemporaryEmailApi2BaseFeature(),
        "test": lambda: TemporaryEmailApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
